demos/https/iot_demo_https_s3_download_async.c

```diff
@@ -774,12 +780,17 @@ static void _readReadyCallback( void * pPrivData,
             }
             else
             {
+                IotLogInfo( "Successfully reconnected to the S3 server." );
```

Comment:
The name of the demo notwithstanding, couldn't this code be used to download from any HTTP server that supports range requests?

Reply:
I suppose so.

Reply:
Updated everywhere to just say "server" instead of "s3 server"